garak/translators/base.py

```diff
@@ -0,0 +1,278 @@
+
+
+class Translator(Configurable):
+    """Base class for objects that execute translation"""
+
+    def __init__(self, config_root: dict = {}) -> None:
+        self._load_config(config_root=config_root)
+
+        self.source_lang, self.target_lang = self.language.split("-")
+
+        self._validate_env_var()
+
+        self._load_translator()
+
+    def _load_translator(self):
+        raise NotImplementedError
+
+    def _translate(self, text: str) -> str:
+        raise NotImplementedError
+
+    def _get_response(self, input_text: str):
+        if self.source_lang is None or self.target_lang is None:
+            return input_text
+
+        translated_lines = []
+
+        split_text = split_input_text(input_text)
+
+        for line in split_text:
+            if self._should_skip_line(line):
+                if contains_invisible_unicode(line):
+                    continue
+                translated_lines.append(line.strip())
+                continue
+            if contains_invisible_unicode(line):
+                continue
+            if len(line) <= 200:
+                translated_lines += self._short_sentence_translate(line)
+            else:
+                translated_lines += self._long_sentence_translate(line)
+
+        return "\n".join(translated_lines)
+
+    def _short_sentence_translate(self, line: str) -> str:
+        translated_lines = []
+        needs_translation = True
+        if self.source_lang == "en" or line == "$":
+            # why is "$" a special line?
+            mean_word_judge = is_english(line)
+            if not mean_word_judge or line == "$":
+                translated_lines.append(line.strip())
+                needs_translation = False
+            else:
+                needs_translation = True
+        if needs_translation:
+            cleaned_line = self._clean_line(line)
+            if cleaned_line:
+                translated_line = self._translate(cleaned_line)
+                translated_lines.append(translated_line)
+
+        return translated_lines
+
+    def _long_sentence_translate(self, line: str) -> str:
+        translated_lines = []
+        sentences = re.split(r"(\. |\?)", line.strip())
+        for sentence in sentences:
+            cleaned_sentence = self._clean_line(sentence)
+            if self._should_skip_line(cleaned_sentence):
+                translated_lines.append(cleaned_sentence)
+                continue
+            translated_line = self._translate(cleaned_sentence)
+            translated_lines.append(translated_line)
+
+        return translated_lines
+
+    def _should_skip_line(self, line: str) -> bool:
+        return (
+            line.isspace()
+            or line.strip().replace("-", "") == ""
+            or len(line) == 0
+            or line.replace(".", "") == ""
+            or line in {".", "?", ". "}
+        )
+
+    def _clean_line(self, line: str) -> str:
+        return remove_english_punctuation(line.strip().lower().split())
```

Comment:
`InjectMorse` may be unhappy with this. Generalisation of issue #371 should solve it (i.e. probes provide character offsets w/ each prompt that denote substrings which should not be permutated by buff/xltn).